.github/workflows/workflow.yaml

```diff
@@ -105,22 +74,25 @@ jobs:
         run: |
           pip install --upgrade build
       - name: Build a binary wheel and a source tarball
+        id: build_pypi
         run: |
           python -m build --sdist --wheel --outdir dist/
       - name: Publish distribution 📦 to Test PyPI
+        if: steps.build_pypi.outcome == 'success' && contains(github.ref, 'refs/tags/') && !contains(github.ref_name, '-')
```

Comment:
Could we add the check for the tag containing a "." as well? Something like ... `&& contains(github.ref_name, '.')`? I think the tag having a "." in it will be more telling than not having a "-" for instance.

Reply:
yes

Reply:
I second Rachel's suggestion. Also, to use `!`, maybe you need to use `${{ }}` https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idif

Reply:
Another option is to add a separate step to check the tag, then store the result and use it when deciding to publish: https://stackoverflow.com/questions/58862864/github-actions-ci-conditional-regex